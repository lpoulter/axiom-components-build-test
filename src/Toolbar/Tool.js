import PropTypes from 'prop-types';
import React, { Component } from 'react';
import ListItem from '../List/ListItem';

export default class Tool extends Component {
  static propTypes = {
    /** Content to be placed inside the Tool */
    children: PropTypes.node.isRequired,
  }

  render() {
    const { children, ...rest } = this.props;

    return (
      <ListItem { ...rest }>
        { children }
      </ListItem>
    );
  }
}
