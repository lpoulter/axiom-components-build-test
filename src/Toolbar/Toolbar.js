import PropTypes from 'prop-types';
import React, { Component } from 'react';
import List from '../List/List';

export default class Toolbar extends Component {
  static propTypes = {
    /** Tools along the Toolbar */
    children: PropTypes.node.isRequired,
  }

  render() {
    const { children } = this.props;

    return (
      <List style="inline">
        { children }
      </List>
    );
  }
}
