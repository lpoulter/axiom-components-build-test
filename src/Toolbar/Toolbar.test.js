import React from 'react';
import renderer from 'react-test-renderer';
import Tool from './Tool';
import Toolbar from './Toolbar';

const getComponent = () =>
  renderer.create(
    <Toolbar>
      <Tool>Test</Tool>
    </Toolbar>
  );

describe('Toolbar', () => {
  it('renders', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
