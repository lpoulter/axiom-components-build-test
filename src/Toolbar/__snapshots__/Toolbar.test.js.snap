// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders 1`] = `
<ul
  className="ax-list ax-list--style-inline ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Test
  </li>
</ul>
`;
